### DISCUSSION TITLE
FirebaseMessaging.onBackgroundMessage() not working for ios

### DISCUSSION THREAD

**Discussion by m2sahin** [firebase/flutterfire#9380] [Q&A]
HI,
I want to use FirebaseMessaging for notifications in my Flutter project. I have completed the installation for both Android and IOS. Everything works perfectly except one thing.

I want to do; I want to evaluate the incoming notification according to the data it contains and decide whether to show the notification or not. I also want to display notifications using flutter_local_notifications package.

I can do all this for android. For iOS, it works fine when the application is running and covered, but the FirebaseMessaging.onBackgroundMessage function is not triggered when the application is terminated.

I've been grappling with this issue for days, there's no advice I haven't tried. Please help me, at least tell me if this is possible. As a result of my research, I realized that there are many people struggling with the same problem and none of them have found a solution. The strange thing is that this issue is never mentioned in firebase and flutter. 

This is very frustrating, please I need help.

**Comment by RokasAwk** [comment]
I can received notification when my app is terminated or in background, but I can't handler the background message on iOS device.

**Reply by m2sahin** [reply]
That's exactly what happens to me.  But the important thing for me is to handle the background notification.
